modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,317 @@
+        3. Include the 'Digest' header and value in the HTTP signature.
+    :param signing_algorithm: The signature algorithm, when signing HTTP requests.
+        Supported values are PKCS1v15, PSS; fips-186-3, deterministic-rfc6979.
+    :param signature_max_validity: The signature max validity,
+        expressed as a datetime.timedelta value.
+    """
+    def __init__(self, key_id, private_key_path,
+                 signing_scheme=SCHEME_HS2019,
+                 signed_headers=[HEADER_CREATED],
+                 signing_algorithm=None,
+                 signature_max_validity=None):
+        """Constructor
+        """
+        self.key_id = key_id
+        """The identifier of the key used to sign HTTP requests.
+        """
+        self.private_key_path = private_key_path
+        """The path of the file containing a private key, used to sign HTTP requests.
+        """
+        self.signing_scheme = signing_scheme
+        """The signature scheme when signing HTTP requests.
+           Supported values are hs2019, rsa-sha256, rsa-sha512.
+        """
+        self.signing_algorithm = signing_algorithm
+        """The signature algorithm when signing HTTP requests.
+           For RSA keys, supported values are PKCS1v15, PSS.
+           For ECDSA keys, supported values are fips-186-3, deterministic-rfc6979.
+        """
+        if signature_max_validity is not None and signature_max_validity.total_seconds() < 0:
+            raise Exception("The signature max validity must be a positive value")
+        self.signature_max_validity = signature_max_validity
+        """The signature max validity, expressed as a datetime.timedelta value.
+        It must be a positive value.
+        """
+        # If the user has not provided any signed_headers, the default must be set to '(created)'.
+        if signed_headers is None or len(signed_headers) == 0:
+            signed_headers = [HEADER_CREATED]
+        if self.signature_max_validity is None and HEADER_EXPIRES in signed_headers:
+            raise Exception(
+                "Signature max validity must be set when " \
+                "'(expires)' signature parameter is specified")
+        if len(signed_headers) != len(set(signed_headers)):
+            raise Exception("Cannot have duplicates in the signed_headers parameter")
+        if HEADER_AUTHORIZATION in signed_headers:
+            raise Exception("'Authorization' header cannot be included in signed headers")
+        self.signed_headers = signed_headers
+        """A list of strings. Each value is the name of HTTP header that must be included
+           in the HTTP signature calculation.
+        """
+        self.private_key = None
+        """The private key used to sign HTTP requests.
+            Initialized when the PEM-encoded private key is loaded from a file.
+        """
+
+    def get_http_signature_headers(self, host, resource_path, method, headers, body, query_params):
```

Comment:
Is host unchanging?
If so, why not pass it in or pass in our configuration instance to this class's `__init__` method?

Reply:
I don't think SDK developers should have to set the host in the HttpSigningConfiguration constructor, because they already do it in the Configuration constructor. But the Configuration class can set the host. This is what I have done locally, you can review when I commit. I will mark resolved for now.